_episodes/l2_02_basic_unit_testing.md

```diff
@@ -0,0 +1,288 @@
+---
+title: "Introduction to Unit Testing using GoogleTest"
+teaching: TBC
+exercises: TBC
+questions:
+- "What framework/libraries can be used for unit testing in C++?"
+- "What is GoogleTest?"
+- "How can I write and run unit tests with GoogleTest?"
+objectives:
+- "Understand the basic components of a unit test"
+- "Understand how to write a unit test using GoogleTest"
+- "Understand how to compile and run the tests"
+- "Understand some of the macros available in GoogleTest"
+keypoints:
+- "We learnt about some of the popular testing frameworks for C++."
+- "We learnt the basics of GoogleTest."
+- "We learnt how to use GoogleTest to write our tests."
+- "We learnt about anatomy of a unit tests." 
+- "We learnt how to write effective unit tests."
+- "Finally, we saw an example to demonstrate the above concepts."
```

Comment:
I think these can do with a rewriting to follow the structure of take away messages we mentioned in other PRs.